tests/test_text.py

```diff
@@ -0,0 +1,811 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: GNU GPLv3
+import numpy as np
+import pandas as pd
+import random
+import sklearn
+from feature.selector import Selective, SelectionMethod
+from feature.text_based import _process_category_data
+from tests.test_base import BaseTest
+from textwiser import TextWiser, Embedding, Transformation
+
+
+class TestText(BaseTest):
+
+    # Verify test usage example
+    def test_usage_example(self):
+        data = pd.DataFrame({"article_1": ["article text here"],
+                             "article_2": ["article text here"],
+                             "article_3": ["article text here"],
+                             "article_4": ["article text here"]})
+        labels = pd.DataFrame({"article_1": [1, 1, 0, 1],
+                               "article_2": [0, 1, 0, 0],
+                               "article_3": [0, 0, 1, 0],
+                               "article_4": [0, 0, 1, 1]},
+                              index=["label_1", "label_2", "label_3", "label_4"])
+
+        # Verify that the number of columns is data and labels match
+        self.assertEqual(data.shape[1], labels.shape[1])
+
+        # Fix the random seed
+        seed = 3000
+        np.random.seed(seed)
+
+        method = SelectionMethod.TextBased(num_features=2,
+                                           optimization_method="random",
+                                           cost_metric="unicost",
+                                           trials=1)
+
+        selector = Selective(method, seed=seed)
+        selector.fit(data, labels)
+        selected_features = selector.transform(data)
+
+        # Verify the selected indices
+        self.assertListEqual(list(selected_features.columns), ['article_1', 'article_3'])
+
+    def test_categorical_data(self):
```

Comment:
@zohrehraziei The only thing that remains is to check that, we can reproduce the examples/tables from the papers. The original problem definition includes "categories" which we flattened out in this library. 

What we need to do is to take a simple example df, that "has" categories, call the _process_category_data() function from the original code and assert that things work.

@bkleyn  can you provide a coverage labels df in the old style to test here? Something small, only to ensure that, given the old process() function, we can run things in this Library. 

In theory, if we run a df with categories (using process) and the same df given as flattened input (as is here) we should get the SAME output. This can be checked here  --and with that we can merge this.

Reply:
I have added the test as per the requirements for README.me. I have also ensured that all tests, including other selection methods, are run to ensure comprehensive testing of the code. 

For df with categories, I have created a simple input df with three articles and three categories. I then called the _process_category_data function with this df, passing in the categories and feature column. I have also added assertions to ensure that the resulting matrix and features have the expected shapes and values.
@bkleyn I kindly request your review of this test and confirmation if it follows the data format used in the paper.